ADR-101: Backport documentation to v0.38.x-experimental (#1477)
cometbft/cometbft: docs/data-companion/pruning.md

---
order: 1
parent:
    title: Pruning Service
    order: 3
---

# Pruning data via the pruning service

CometBFT employs a sophisticated pruning logic to eliminate unnecessary data and reduce storage requirements.

This document covers use cases where the pruning process on a CometBFT node can be influenced via the Data Companion
pruning service API.

> The pruning process runs asynchronously in the background based on the retain

Reviewer: I would start with something like:

"Starting from this release, ..."

> heights set either by the application or via the Data Companion API. Before the

Reviewer: ```suggestion
heights set either by the application or via the Data Companion pruning service API. Before the
```

> Data Companion API was introduced, that is, before v0.38, pruning could be

Reviewer: For all effects, this is v0.38, right? Moreover, this is not enabled in the actual (not experimental) v0.38.

Reviewer 2: ```suggestion
Data Companion API was introduced, that is, in v0.38 and earlier versions, pruning could be
```

triggered only by the application at some height, and CometBFT would not advance

> to the next height until the blocks were pruned. Now, the application just need

Reviewer: ```suggestion
to the next height until the blocks were pruned. The pruning service was therefore synchronous. Now, the application just need
```

Reviewer 2: ```suggestion
to the next height until the blocks were pruned. In this experimental version of v0.38.x, the application just need
```

> to set the desired retain heights, and then consensus and pruning run in

Reviewer: ```suggestion
to set the desired retain heights, and the pruning service runs in parallel with consensus.
```

This document covers use cases where the pruning process on a CometBFT node can be influenced via the Data Companion
pruning service API.

Starting from this release, the pruning process runs asynchronously in the
background based on the retain heights set either by the application or via the
Data Companion pruning service API. Before the Data Companion API was
introduced, that is, in v0.38 and earlier versions, pruning could be triggered
only by the application at some height, and CometBFT would not advance to the
next height until the blocks were pruned. The pruning service was therefore

> synchronous. In this experimental version of v0.38.x, the application just need

Reviewer 2: ```suggestion
synchronous. In this experimental version of v0.38.x, the application just needs
```
